src/components/IGVBrowser.js

```diff
@@ -1,80 +1,115 @@
-        const [bucket, file] = parseGsUri(filePath)
-        const knownBucketStatus = knownBucketRequesterPaysStatuses.get()[bucket]
-
-        if (knownBucketStatus !== undefined) {
-          return knownBucketStatus
-        } else {
-          try {
-            await Ajax(signal).Buckets.getObject(workspace.workspace.googleProject, bucket, file, { fields: 'kind' })
-            return false
-          } catch (e) {
-            if (e.requesterPaysError) {
-              return true
-            }
+  const addTracks = withErrorReporting('Unable to add tracks', async tracks => {
+    // Select one file per each bucket represented in the tracks list.
+    const bucketExemplars = _.flow(
+      _.map(_.get('url')),
+      _.uniqBy(url => {
+        const [bucket] = parseGsUri(url)
+        return bucket
+      })
+    )(tracks)
+
+    // Learn the requester pays status of each bucket.
+    // Requesting a file will store its requester pays status in knownBucketRequesterPaysStatuses.
+    const isRequesterPays = await Promise.all(_.map(async url => {
+      const [bucket, file] = parseGsUri(url)
+
+      if (knownBucketRequesterPaysStatuses.get()[bucket] === undefined) {
+        try {
+          await Ajax(signal).Buckets.getObject(workspace.workspace.googleProject, bucket, file, { fields: 'kind' })
+        } catch (e) {
+          if (!e.requesterPaysError) {
+            throw e
           }
         }
-      }, fileBucketExemplars))
+      }
+      return knownBucketRequesterPaysStatuses.get()[bucket]
+    }, bucketExemplars))
 
-      if (!requesterPaysProjectStore.get() && _.some(_.identity, bucketRpStatuses)) {
-        onRequesterPaysError()
-      } else {
-        try {
-          const { default: igv } = await import('igv')
-          igvLibrary.current = igv
-
-          const options = {
-            genome,
-            reference,
-            tracks: await Promise.all(_.map(async ({ filePath, indexFilePath }) => {
-              const [bucket] = parseGsUri(filePath)
-              const userProjectParam = { userProject: knownBucketRequesterPaysStatuses.get()[bucket] ? await getUserProjectForWorkspace(workspace) : undefined }
-
-              return {
-                name: `${_.last(filePath.split('/'))} (${filePath})`,
-                url: Utils.mergeQueryParams(userProjectParam, filePath),
-                indexURL: !!indexFilePath ? Utils.mergeQueryParams(userProjectParam, indexFilePath) : undefined
+    // If any bucket is requester pays, files in that bucket will need to have a user project included in the request.
+    let userProject
+    if (_.some(_.identity, isRequesterPays)) {
+      // Check if the user can bill to the current workspace.
+      userProject = await getUserProjectForWorkspace(workspace)
+
+      // If not, prompt to select a workspace to bill to.
+      if (!userProject) {
+        userProject = await new Promise((resolve, reject) => {
```

Comment:
I couldn't tell why `userProject` had to be a non-`const` since it doesn't seem to be used besides in `if (!userProject)`. Could it be

```suggestion
    if (_.some(_.identity, isRequesterPays)) {
      // Check if the user can bill to the current workspace.
      const userProject = await getUserProjectForWorkspace(workspace)

      // If not, prompt to select a workspace to bill to.
      if (!userProject) {
        await new Promise((resolve, reject) => {
```
?

Reply:
`userProject` is used to add a query parameter to requests for files in requester pays buckets.
```js
 _.forEach(({ name, url, indexURL }) => {
  const [bucket] = parseGsUri(url)
  const userProjectParam = { userProject: knownBucketRequesterPaysStatuses.get()[bucket] ? userProject : undefined }
  ...
```

If we can't automatically determine a user project using `getUserProjectForWorkspace`, we have to prompt the user to select one.

Reply:
Ah okay. I'd missed that somehow.